### PR TITLE
Allow reverse relationship direction to be accessed from the API

### PR DESCRIPTION
**Pull request by adgad**
/cc @Financial-Times/next-myft 

Opens up the API to query all node types (not just user). e.g. `/v1/content/028c2eaa-668c-11e5-a57f-21b88f7d973f/saved` to get a list of users who saved that particular article.

* In theory, this enables writes to those endpoints too. Can't see a use for this yet though, so perhaps it should be discouraged/disabled?
* Caching is only enabled on /user/ endpoints

This PR will allow us to migrate over the few bits of the stack that are still using the old API:
- [ ] Push notifications (next-content-notifier) - needs to get all endpoints, and all users following a topic 
- [ ] Emails (next-myft-email-generator) - needs to get a list of users signed up to daily/weekly email

### CODE REVIEW

File: server/controllers/relationship/update.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var purgeCache = require('../../lib/purge-cache');
 
 module.exports = (req, res, next) => {
 
-	if(whitelist.nodes.indexOf(req.params.node)) {
+	if(whitelist.nodes.indexOf(req.params.node) < 0) {
```
Comment (phamann) [inline server/controllers/relationship/update.js]: Potentially abstract this logic into the `whitelist` module i.e. `whitelist.isNotValid(node)` as we're repeating a lot.

File: server/models/crud.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Crud {
 			}
 
 			const query = `MATCH (actor:${this.type} { uuid: {uuid}})
-							OPTIONAL MATCH (actor)-[relationship]->(subject)
+							OPTIONAL MATCH (actor)<-[relationship]->(subject)
```
Comment (phamann) [inline server/models/crud.js]: I thought `<-[relationship]->` didn't work?

File: server/models/crud.js [hunk 3]
```diff
@@ -276,9 +300,12 @@ class Crud {
 				params[key] = body[key];
 				return `SET rel.${key} = {${key}}`;
 			}).join(' \n');
+
+			const directionalRel = direction === 'in' ? `<-[rel:${model.label}]-` : `-[rel:${model.label}]->`;
```
Comment (phamann) [inline server/models/crud.js]: Might not be worth it, but you could also abstract this logic inside this file:
```javascript
function directionRelationshipCypher(label, direction) {
    return direction === 'in' ? `<-[rel:${label}]-` : `-[rel:${label}]->`;
}

**Comment by phamann** [comment]
:+1: Looks good to me :)

File: server/models/crud.js (hunk 2 above)
Comment (adgad) [inline server/models/crud.js]: It acts as an OR rather than an AND when querying

Comment (phamann) [inline server/models/crud.js]: Nice!

**Comment by adgad** [comment]
Made a couple of tidy up DRY changes

**Pull request merged**